### PR TITLE
kitty: 0.28.1 -> 0.29.0

### PR DESCRIPTION
**Pull request by adamcstephens**
###### Description of changes

https://github.com/kovidgoyal/kitty/blob/v0.29.0/docs/changelog.rst#0290-2023-07-10

Also includes a fix to resolve vendorHash not updating correctly with nix-update.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/kitty/default.nix [hunk 1]
```diff
@@ -29,16 +29,16 @@
...
 
   src = fetchFromGitHub {
     owner = "kovidgoyal";
     repo = "kitty";
     rev = "refs/tags/v${version}";
-    hash = "sha256-pAo+bT10rdQOf9j3imKWCCMFGm8KntUeTQUrEE1wYZc=";
+    hash = "sha256-FTitj43RFCNvSWInXHALyIljfcBBkaq/XI1ZA1k0glk=";
   };
-  vendorHash = "sha256-vq19exqsEtXhN20mgC5GCpYGm8s9AC6nlfCfG1lUiI8=";
+  vendorHash = "sha256-jk2EcYVuhV/UQfHAIfpnn8ZIZnwjA/o8YRXmpoC85Vc=";
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/kitty/default.nix:41]:
```suggestion
```

File: pkgs/applications/terminal-emulators/kitty/default.nix [hunk 2]
```diff
@@ -100,10 +100,12 @@ buildPythonApplication rec {
   CGO_ENABLED = 0;
   GOFLAGS = "-trimpath";
 
+  # allow for both overrideAttrs and vendorHash updating
   goModules = (buildGoModule {
     pname = "kitty-go-modules";
     inherit src vendorHash version;
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/kitty/default.nix:106]:
```suggestion
    inherit src version;
    vendorHash = "sha256-jk2EcYVuhV/UQfHAIfpnn8ZIZnwjA/o8YRXmpoC85Vc=";
```

Comment (adamcstephens) [reply pkgs/applications/terminal-emulators/kitty/default.nix]: I made this change, but also moved `goModules` up by the source so it's easier to see if someone wants to override it.

File: pkgs/applications/terminal-emulators/kitty/default.nix [hunk 3]
```diff
@@ -100,10 +100,12 @@ buildPythonApplication rec {
...
   }).go-modules;
+  go-modules = goModules;
```
[the remark this answers is not in the document]
Comment (adamcstephens) [reply pkgs/applications/terminal-emulators/kitty/default.nix]: Thanks, I like that.

File: pkgs/applications/terminal-emulators/kitty/default.nix [hunk 4]
```diff
@@ -29,16 +29,21 @@
...
-    hash = "sha256-pAo+bT10rdQOf9j3imKWCCMFGm8KntUeTQUrEE1wYZc=";
+    hash = "sha256-FTitj43RFCNvSWInXHALyIljfcBBkaq/XI1ZA1k0glk=";
   };
-  vendorHash = "sha256-vq19exqsEtXhN20mgC5GCpYGm8s9AC6nlfCfG1lUiI8=";
+
+  goModules = (buildGoModule {
+    pname = "kitty-go-modules";
+    inherit src version;
+    vendorHash = "sha256-jk2EcYVuhV/UQfHAIfpnn8ZIZnwjA/o8YRXmpoC85Vc=";
+  }).go-modules;
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/kitty/default.nix:46]:
```suggestion
  inherit = (buildGoModule {
    pname = "kitty-go-modules";
    inherit src version;
    vendorHash = "sha256-jk2EcYVuhV/UQfHAIfpnn8ZIZnwjA/o8YRXmpoC85Vc=";
  }) go-modules;
```
naive question: why don't we do this and remove go-modules from passthru and rename the bash variable below?

Comment (adamcstephens) [reply pkgs/applications/terminal-emulators/kitty/default.nix:46]: Are you suggesting to effectively revert https://github.com/NixOS/nixpkgs/pull/241707 ?

**Comment by SuperSandro2000** [comment]
darwin tests are unhappy....

```
======================================================================
ERROR: test_bash_integration (kitty_tests.shell_integration.ShellIntegrationWithKitten) (argv='bash -s arg1 --rcfile rcfile')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/nix-build-kitty-0.29.0.drv-0/source/kitty/launcher/kitty.app/Contents/MacOS/../../../../../kitty_tests/shell_integration.py", line 352, in run_test
    pty.wait_till(lambda: wait_string in pty.screen_contents())
    argv = 'bash -s arg1 --rcfile rcfile'
    assert_not_in = True
    excluded = ()
    expected = ('rcfile',)
    pty = <kitty_tests.PTY object at 0x1081a05e0>
    rc = 'echo ok;read'
    self = <kitty_tests.shell_integration.ShellIntegrationWithKitten testMethod=test_bash_integration>
    setup_env = <function ShellIntegration.test_bash_integration.<locals>.setup_env at 0x1090d0af0>
    wait_string = 'ok'
  File "/private/tmp/nix-build-kitty-0.29.0.drv-0/source/kitty/launcher/kitty.app/Contents/MacOS/../../../../../kitty_tests/__init__.py", line 296, in wait_till
    raise TimeoutError(f'The condition was not met. Screen contents: \n {repr(self.screen_contents())}')
    end_time = 16339622.9105604
    q = <function ShellIntegration.test_bash_integration.<locals>.run_test.<locals>.<lambda> at 0x1091cd510>
    self = <kitty_tests.PTY object at 0x1081a05e0>
    timeout = 10
TimeoutError: The condition was not met. Screen contents: 
 '[profile]\n[.bash_profile]\nPROMPT $ '
======================================================================
FAIL: test_bash_integration (kitty_tests.shell_integration.ShellIntegrationWithKitten) (argv='bash')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/nix-build-kitty-0.29.0.drv-0/source/kitty/launcher/kitty.app/Contents/MacOS/../../../../../kitty_tests/shell_integration.py", line 358, in run_test
    self.assertIn(f'[{x}]', q)
    argv = 'bash'
    assert_not_in = False
    excluded = ()
    expected = ('bash.bashrc', '.bashrc')
    pty = <kitty_tests.PTY object at 0x1088f73a0>
    q = '[profile]\n[.bash_profile]\nPROMPT $ '
    rc = ''
    self = <kitty_tests.shell_integration.ShellIntegrationWithKitten testMethod=test_bash_integration>
    setup_env = <function ShellIntegration.test_bash_integration.<locals>.setup_env at 0x1090d0af0>
    wait_string = 'PROMPT $'
    x = 'bash.bashrc'
AssertionError: '[bash.bashrc]' not found in '[profile]\n[.bash_profile]\nPROMPT $ '
======================================================================
FAIL: test_bash_integration (kitty_tests.shell_integration.ShellIntegrationWithKitten) (argv='bash --rcfile rcfile')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/nix-build-kitty-0.29.0.drv-0/source/kitty/launcher/kitty.app/Contents/MacOS/../../../../../kitty_tests/shell_integration.py", line 358, in run_test
    self.assertIn(f'[{x}]', q)
    argv = 'bash --rcfile rcfile'
    assert_not_in = False
    excluded = ()
    expected = ('bash.bashrc', 'rcfile')
    pty = <kitty_tests.PTY object at 0x108f21a80>
    q = '[profile]\n[.bash_profile]\nPROMPT $ '
    rc = ''
    self = <kitty_tests.shell_integration.ShellIntegrationWithKitten testMethod=test_bash_integration>
    setup_env = <function ShellIntegration.test_bash_integration.<locals>.setup_env at 0x1090d0af0>
    wait_string = 'PROMPT $'
    x = 'bash.bashrc'
AssertionError: '[bash.bashrc]' not found in '[profile]\n[.bash_profile]\nPROMPT $ '
======================================================================
FAIL: test_bash_integration (kitty_tests.shell_integration.ShellIntegrationWithKitten) (argv='bash --init-file rcfile')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/nix-build-kitty-0.29.0.drv-0/source/kitty/launcher/kitty.app/Contents/MacOS/../../../../../kitty_tests/shell_integration.py", line 358, in run_test
    self.assertIn(f'[{x}]', q)
    argv = 'bash --init-file rcfile'
    assert_not_in = False
    excluded = ()
    expected = ('bash.bashrc', 'rcfile')
    pty = <kitty_tests.PTY object at 0x10807af80>
    q = '[profile]\n[.bash_profile]\nPROMPT $ '
    rc = ''
    self = <kitty_tests.shell_integration.ShellIntegrationWithKitten testMethod=test_bash_integration>
    setup_env = <function ShellIntegration.test_bash_integration.<locals>.setup_env at 0x1090d0af0>
    wait_string = 'PROMPT $'
    x = 'bash.bashrc'
AssertionError: '[bash.bashrc]' not found in '[profile]\n[.bash_profile]\nPROMPT $ '
======================================================================
FAIL: test_bash_integration (kitty_tests.shell_integration.ShellIntegrationWithKitten)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/nix-build-kitty-0.29.0.drv-0/source/kitty/launcher/kitty.app/Contents/MacOS/../../../../../kitty_tests/shell_integration.py", line 381, in test_bash_integration
    self.ae(env.get('ES'), 'a\n `b` c\n$d', f'Screen contents: {pty.screen_contents()!r}')
    env = {'BASH_SILENCE_DEPRECATION_WARNING': '1', 'KITTY_INSTALLATION_DIR': '/private/tmp/nix-build-kitty-0.29.0.drv-0/source', 'LANG': 'en_US.UTF-8'}
    ps1 = 'line1\nprompt> '
    pty = <kitty_tests.PTY object at 0x108f22dd0>
    q = '/private/tmp/nix-build-kitty-0.29.0.drv-0/tmpgvmpel70/testing-cwd-notification-🐱'
    redrawn = <function ShellIntegration.test_bash_integration.<locals>.redrawn at 0x108fd2cb0>
    run_test = <function ShellIntegration.test_bash_integration.<locals>.run_test at 0x1090d09d0>
    self = <kitty_tests.shell_integration.ShellIntegrationWithKitten testMethod=test_bash_integration>
    setup_env = <function ShellIntegration.test_bash_integration.<locals>.setup_env at 0x1090d0af0>
AssertionError: None != 'a\n `b` c\n$d' : Screen contents: '[profile]\n[.bash_profile]\nclone-in-kitty\nPROMPT $ clone-in-kitty\nPROMPT $ '
----------------------------------------------------------------------
```

**Comment by adamcstephens** [comment]
Darwin tests "fixed".

**Pull request merged**